webapp/src/ts/modules/contacts/contacts-report.component.ts

```diff
@@ -0,0 +1,243 @@
+    this.globalActions.setEnketoEditedStatus(true);
+  }
+
+  private resetFormError() {
+    if (this.enketoError) {
+      this.globalActions.setEnketoError.setEnketoError(null);
+    }
+  }
+
+  private render() {
+    this.contentError = false;
+    this.errorTranslationKey = false;
+
+    return this
+      .getContact()
+      .then((contact) => {
+        this.setCancelCallback();
+        this.selectedContact = contact;
+        return this.xmlFormsService.get(this.routeSnapshot.params.formId);
+      })
+      .then((form) => {
+        this.globalActions.setEnketoEditedStatus(false);
+        this.globalActions.setTitle(this.translateFromService.get(form.title));
+        this.setCancelCallback();
+
+        const instanceData = {
+          source: 'contact',
+          contact: this.selectedContact.doc,
+        };
+        const markFormEdited = this.markFormEdited.bind(this);
+        const resetFormError = this.resetFormError.bind(this);
+
+        return this.enketoService.render(
+          '#contact-report',
+          form,
+          instanceData,
+          markFormEdited,
+          resetFormError
+        );
+      })
+      .then((formInstance) => {
+        this.form = formInstance;
+        this.loadingForm = false;
+        this.telemetryData.postRender = Date.now();
+        this.telemetryData.form = this.routeSnapshot.params.formId;
+
+        this.telemetryService.record(
+          `enketo:contacts:${this.telemetryData.form}:add:render`,
+          this.telemetryData.postRender - this.telemetryData.preRender);
+      })
+      .catch(err => {
+        console.error('Error loading form', err);
+        this.errorTranslationKey = err.translationKey || 'error.loading.form';
+        this.contentError = true;
+        this.loadingForm = false;
+      });
+  }
+
+  private subscribeToStore() {
+    const reduxSubscription = combineLatest(
+      this.store.select(Selectors.getEnketoStatus),
+      this.store.select(Selectors.getEnketoSavingStatus),
+      this.store.select(Selectors.getEnketoError),
+    ).subscribe(([ enketoStatus, enketoSaving, enketoError]) => {
+      this.enketoStatus = enketoStatus;
+      this.enketoSaving = enketoSaving;
+      this.enketoError = enketoError;
+    });
+    this.subscription.add(reduxSubscription);
+  }
+
+  private subscribeToRoute() {
+    this.routeSnapshot = this.route.snapshot;
+    this.resetState();
+
+    const routeSubscription = this.route.params.subscribe((params) => {
+      if (_isEqual(this.routeSnapshot.params, params)) {
+        // the 1st time we load the form, we must wait for the view to be initialized
+        // if we don't skip, it will result in the form being loaded twice
+        return;
+      }
+      this.routeSnapshot = this.route.snapshot;
+      this.resetState();
+      this.resetFormError();
+      this.render();
+    });
+    this.subscription.add(routeSubscription);
+
+    const queryParamsSubscription = this.route.queryParams.subscribe(() => {
```

Comment:
Is it possible to get queryParams here? Which query params are you expecting?

Reply:
Not necessary. I have removed the code block.